notebooks/Interpretability - Image Explainers.ipynb

```diff
@@ -0,0 +1 @@
.setTargetClassesCol(\"top2pred\")\n\nshap_result = shap.transform(predicted)\\\n  .withColumn(\"shaps_piano\", col(\"shaps\").getItem(0))\\\n  .withColumn(\"shaps_cello\", col(\"shaps\").getItem(1))\\\n  .withColumn(\"r2_piano\", vec_access(\"r2\", lit(0)))\\\n  .withColumn(\"r2_cello\", vec_access(\"r2\", lit(1)))\\\n  .cache()\n\ndisplay(shap_result.select(col(\"shaps_piano\"), col(\"r2_piano\"), col(\"shaps_cello\"), col(\"r2_cello\")))\nshap_row = shap_result.head()"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"23cf13f4-093e-441d-becf-29c8d1eead0e"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"data":"","errorSummary":"","metadata":{},"errorTraceType":null,"type":"ipynbError","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>"]}}],"execution_count":0},{"cell_type":"markdown","source":["We plot the SHAP values for \"piano\" output and \"cell\" output.\n\nGreen area are superpixels with SHAP values above 90 percentile, and red area are superpixels with SHAP values below 10 percentile.\n\n> Notice that we drop the base value from the SHAP output before rendering the superpixels. The base value is the model output for the background (all black) image."],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"d069a2bd-20f2-4235-93a3-b145f93e2b4e"}}},{"cell_type":"code","source":["plot_superpixels(barr, shap_row['superpixels']['clusters'], list(shap_row['shaps_piano'][1:]))\nplot_superpixels(barr, shap_row['superpixels']['clusters'], list(shap_row['shaps_cello'][1:]))"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"63c8d593-e336-46d5-94f4-a7466b65bfc7"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"data":"","errorSummary":"","metadata":{},"errorTraceType":null,"type":"ipynbError","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>"]}}],"execution_count":0}],"metadata":{"application/vnd.databricks.v1+notebook":{"notebookName":"Interpretability - Image Explainers","dashboards":[],"notebookMetadata":{"pythonIndentUnit":2},"language":"python","widgets":{},"notebookOrigID":496759265154124}},"nbformat":4,"nbformat_minor":0}
```

Comment:
nit: if theres a way to expand this as lines it will make it easier to comment on. I think you should perhaps include a still of the expected result so that even in unrendered NB people can get the gist of what it does ;)

Reply:
Done. I reformatted the notebooks, and added some visuals to demonstrate the expected results.